src/Term/TermFallback.php

```diff
@@ -46,6 +47,28 @@ public function __construct( $requestedLanguageCode, $text, $actualLanguageCode,
 			throw new InvalidArgumentException( '$sourceLanguageCode must be a non-empty string or null' );
 		}
 
+		if ( $sourceLanguageCode === null && $actualLanguageCode === $requestedLanguageCode ) {
```

Comment:
No, 3 variables  means 3 combinations of how 2 can be equal. The comparison with null is && not ||.